File: newton-school-course/triplet-sum-ada-contest-1-set-1-adypu-25-29/jv9byxkvn4so.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let inputLines = [];
rl.on('line', (line) => {
    inputLines.push(line.trim());
});

rl.on('close', () => {
    if (inputLines.length === 0) return;
    
    let lineIdx = 0;
    let t = parseInt(inputLines[lineIdx++]);
    
    let results = [];
    
    while (t > 0 && lineIdx < inputLines.length) {
        if (inputLines[lineIdx] === "") {
            lineIdx++;
            continue;
        }
        
        let [n, target] = inputLines[lineIdx++].split(/\s+/).map(Number);
        let nums = inputLines[lineIdx++].split(/\s+/).map(Number);
        
        let found = false;
        // Check all possible triplets (i, j, k)
        for (let i = 0; i < n; i++) {
            for (let j = i + 1; j < n; j++) {
                for (let k = j + 1; k < n; k++) {
                    if (nums[i] + nums[j] + nums[k] === target) {
                        found = true;
                        break;
                    }
                }
                if (found) break;
            }
            if (found) break;
        }
        
        results.push(found ? "YES" : "NO");
        t--;
    }
    
    console.log(results.join('\n'));
});
for _ in range(t):
    # Write your code here

File: newton-school-course/triplet-sum-ada-contest-1-set-1-adypu-25-29/jv9byxkvn4so.py
import sys

def solve():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    t = int(input_data[0])
    idx = 1
    
    results = []
    for _ in range(t):
        if idx >= len(input_data):
            break
        n = int(input_data[idx])
        target = int(input_data[idx+1])
        idx += 2
        
        nums = []
        for _ in range(n):
            nums.append(int(input_data[idx]))
            idx += 1
            
        found = False
        # Brute-force through all triplets (i, j, k)
        for i in range(n):
            for j in range(i + 1, n):
                for k in range(j + 1, n):
                    if nums[i] + nums[j] + nums[k] == target:
                        found = True
                        break
                if found:
                    break
            if found:
                break
                
        results.append("YES" if found else "NO")
        
    print('\n'.join(results))

if __name__ == '__main__':
    solve()